src/main/resources/META-INF/rewrite/spring-boot-23.yml

```
@@ -39,6 +39,12 @@ recipeList:
       groupId: org.springframework.boot
       artifactId: spring-boot-starter-parent
       newVersion: 2.3.x
+  # https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-2.3-Release-Notes#validation-starter-no-longer-included-in-web-starters
+  - org.openrewrite.maven.AddDependency:
+      groupId: org.springframework.boot
+      artifactId: spring-boot-starter-validation
+      version: 2.3.x
+      onlyIfUsing: javax.validation.constraints.*
 
```

Might be best to [add this new flag](https://github.com/openrewrite/rewrite/blob/9ecb9a1f75b9f4a8289e97f9fd6ed5e2a930f008/rewrite-maven/src/main/java/org/openrewrite/maven/AddDependency.java#L128-L133) to prevent this from being added if it's also available transitively.
```suggestion
      acceptTransitive: true

```